packages/mu-transport/index.js

```diff
@@ -39,17 +40,47 @@ module.exports = function transport (createDriver, mu, opts) {
     tearDown: driver.tearDown
   }
 
+
+
+  function constructFailResponse (err, msg) {
+    var pkt = { err: err,
+                response: {},
+                protocol: {}}
+    var elt = muid
+
+    /* istanbul ignore else  */
```

Comment:
I submit that its worth the investment for core - the fiddling can reveal things not thought of, or if not it gives confidence that the logic is sound and cases well covered